Source/JavaScriptCore/b3/B3LowerToAir.cpp

```diff
@@ -430,13 +430,14 @@ class LowerToAir {
     {
         Tmp& tmp = m_valueToTmp[value];
         if (!tmp) {
+            Value* prevValue = value;
             while (shouldCopyPropagate(value))
                 value = value->child(0);
 
             if (value->opcode() == FramePointer)
                 return Tmp(GPRInfo::callFrameRegister);
 
-            Tmp& realTmp = m_valueToTmp[value];
+            Tmp& realTmp = prevValue == value ? tmp : m_valueToTmp[value];
             if (!realTmp) {
```

Comment:
I think this `tmp` is not so costly, right? If so, the original code looks simpler and easier to read.